discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -515,3 +529,72 @@ def wait_for_cluster_to_be_in_installing_status(self):
             cluster_id=self.id,
             statuses=[consts.ClusterStatus.INSTALLING]
         )
+
+    def reset_cluster_and_wait_for_ready(self, cluster, nodes):
+        # Reset cluster install
+        cluster.reset_install()
+        assert cluster.is_in_insufficient_status()
+        # Reboot required nodes into ISO
+        cluster.reboot_required_nodes_into_iso_after_reset(nodes=nodes)
+        # Wait for hosts to be rediscovered
+        cluster.wait_until_hosts_are_discovered()
+        cluster.wait_for_ready_to_install()
+
+    def verify_no_logs_uploaded(self, cluster, path):
+        with pytest.raises(ApiException) as ex:
+            cluster.download_installation_logs(path)
+        assert "No log files" in str(ex.value)
+
+    def verify_logs_uploaded(self, path, expected_min_log_num, installation_success):
+
+        # verify that logs were collected from all expected sources
+        assert (os.path.exists(path))
+        dir_path = path.split(".")[0]
+        try:
+            tar = tarfile.open(path)
+            logging.info(f'downloaded logs: {tar.getnames()}')
+            assert len(tar.getnames()) >= expected_min_log_num
+            tar.extractall(dir_path)
+            for gz in os.listdir(dir_path):
+                if "bootstrap" in gz:
+                    self._verify_node_logs_uploaded(dir_path, gz)
+                    self._verify_bootstrap_logs_uploaded(dir_path, gz, installation_success)
+                elif "master" in gz or "worker" in gz:
+                    self._verify_node_logs_uploaded(dir_path, gz)
+            tar.close()
+        finally:
+            # clean up
+            path = os.path.abspath(path)
+            os.remove(path)
```

Comment:
Please don't delete the logs you got for verification.

Reply:
Same, this is existing code, we can change it in a different PR.

Reply:
Logs should be deleted, test teardown collect logs in case of failure so it wont be lost.